unit-tests/py/common.py

```diff
@@ -0,0 +1,117 @@
+        exit(p.returncode)
+
+# Returns the first device that was found, if no device is found the test is aborted
+def get_first_device():
+    c = rs.context()
+    if not c.devices.size():  # if no device is connected we don't run the AC test
+        print("No device found, aborting AC test")
+        exit(0)
+    return c.devices[0]
+
+# Functions for asserting test cases:
+# Receives an expression which is an assertion. If false the assertion failed
+def require(exp):  
+    global n_assertions, n_failed_assertions, test_failed
+    n_assertions += 1
+    if not exp:
+        n_failed_assertions += 1
+        test_failed = True
+        print("Require failed")
+        for line in traceback.format_stack():
+            print(line.strip())
+
+# Receives the resulted value and the expected value and asserts they are equal
+def require_equal(result, expected):
+    global n_assertions, n_failed_assertions, test_failed
+    n_assertions += 1
+    if not result == expected:
+        n_failed_assertions += 1
+        test_failed = True
+        print("Result was: " + result + "\nBut we expected: " + expected)
+        for line in traceback.format_stack():
+            print(line.strip())
+
+# This function should never be reached, it receives the number of assertion that were 
+# skipped if it was reached
+def require_no_reach(skipped):
+    require(False)
+
+# Receives 2 lists and asserts they are identical
+def require_equal_lists(result, expected):
+    global n_assertions, n_failed_assertions, test_failed
+    n_assertions += 1
+    failed = False
+    if not len(result) == len(expected):
+        n_failed_assertions += 1
+        test_failed = True
+        failed = True
+        print("Require equal lists failed due to lists of different sizes:")
+        print("The resulted list has " + str(len(result)) + " elements, but the expected list has " + str(len(expected)) + " elements")
+    i = 0
+    for res, exp in zip(result, expected):
+        n_assertions += 1
+        if not res == exp:
+            n_failed_assertions += 1
+            test_failed = True
+            failed = True
+            print("Require equal lists failed due to unequal elements:")
+            print("The element of index " + str(i) + " in both lists was not equal")
+        i += 1
+    if failed:
+        print("Result list:", end = ' ')
+        print(result)
+        print("Expected list:", end = ' ')
+        print(expected)
+        for line in traceback.format_stack():
+            print(line.strip())
+
+# Receives an exception and asserts its type and message is as expected
+def require_exception(exception, expected_type, expected_msg = ""):
+    require_equal(type(exception), expected_type)
+    if expected_msg:
+        require_equal(str(exception), expected_msg)
+
+# Functions for formatting test cases
+def start(msg):
+    global n_tests, test_failed
+    n_tests += 1
+    test_failed = False
+    print(msg)
```

Comment:
May want to require that the previous test finished...

Reply:
Not sure how to do that, and don't see how it's possible it won't be finished. Tests run one after the other